lib/engine/step/train.rb

```diff
@@ -90,8 +58,14 @@ def process_buy_train(action)
 
         @log << "#{entity.name} #{verb} a #{train.name} train for "\
           "#{@game.format_currency(price)} from #{source}"
+        @round.trains_bought << {
+          entity: entity,
+          train: train,
+          from_depot?: train.from_depot?,
```

Comment:
do you need all these params? it doesn't seem like you use them all

Reply:
why don't we just put this where it's necessary, the single buy, i don't think you need to track all this stuff, just whether or not you bought a train right?

Reply:
Have now changed so that trains_bought is only updated by special_train_buy and single_depot_train_buy_before_phase4.

The information is now just entity, which seems to be the only one needed. (Was not sure when the @round state was reset, therefor the extra values.) 
If adding train.sym it would be possible to refactor 1836Jr30 to use this state as well, but in that case one could consider to add methods in train.rb to handle this in a general way.